## Cuarta actividad evaluativa

Profesor Alfonso Cervantes Barragán - 05-10-2020

Universidad Sergio Arboleda - Informatica III

### Descripción del contexto.

En esta actividad el estudiante debera realizar un análisis exploratorio de Datos (EDA) del dataset *poblacion_prision_india.csv*, que contiene información sobre la población de prisioneros en la India de los años 2001 al 2013.  

La descripción de las columnas del dataset es la siguiente:

|Columna |	Descripción |  
|--------|--------------|
|States_UT_Name	|Estado en donde se llevo el censo |   
|year| año del censo|   
|Jail_type|	Tipo de institución  | 
|year |	año de la observación|   
|Male Convicts | Convictos masculinos|  
|Female_Convicts |	Convictos femeninos|  
|Total_Convicts|	Número total de convictos |  
|Male_under_trial| 	Hombres bajo juicio  | 
|Femail_under_trial| Mujeres bajo juicio | 
|Total_under_trial| Total de personas bajo juicio|
|Male_detenues| detenidos masculinos|
|Female_detenues|detenidos femeninos|
|Total_detenues| total de detenidos|
|Male_Others| ocupantes masculinos bajo otra razón|
|Female_others|ocupantes femeninos bajo otra razón|
|Total_others|Número total de ocupantes bajo otra razón|
|Total_male|Número total de ocupantes masculinos|
|Total_female|Número total de ocupantes femeninos|
|Grand_total| Número total de ocupantes|



#### Entregable

El estudiante debera entregar un documento con extensión *ipynb* mediante el taller "Cuarta Evaluación " que se encuentra en el curso virtual Informatica III en el plataforma Sergio Virtual


#### Plazo

El plazo de entrega sera hasta el sabado, 24 de octubre a las 11:30 p.m. En el aula del curso en la plataforma Sergio Virtual estará disponible un taller en el cual deberan subir el archivo .ipynb (Jupyter notebook) con todo el código requerido. 

No esta permitido resolver los ejercicios a mano ni ingresar la respuesta sin que haya un código que lo respalde.

#### Donde puedo trabajar?

Este ejercicio se puede realizar en el entorno virtual creado con base en el repositorio github.com/alfonsocerv/datascience en la carpeta /ejericicio/caso10.

#### Criterios de evaluación

1. Todo el código requerido debe estar presente. Cada ejercicio debe tener su respectiva respuesta.
2. No pueden haber errores que se generen al ejecutar el código.
3. El código debe cumplir con las especificaciones de cada ejercicio.

### Ejercicio 1

#### Importa las librerias pandas, numpy y matplotlib


In [4]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [5]:
#cargamos la base de datos
data = pd.read_csv("poblacion_prision_india.csv")

### Ejercicio 2

#### Queremos saber de manera general de qué se trata esta base de datos



In [6]:
data.head()

State_UT_name  Year       Jail_Type  Male Convicts  Female_Convicts  \
0  Andhra Pradesh  2001    Central Jail           3372                3   
1  Andhra Pradesh  2001   District Jail            294                1   
2  Andhra Pradesh  2001        Sub_Jail            188               10   
3  Andhra Pradesh  2001      Women Jail              0               88   
4  Andhra Pradesh  2001  Borstal School             17                0   

   Total_Convicts  Male_Under_trial  Female_Under_trial  Total_Under_trial  \
0            3375              2718                 114               2832   
1             295              2379                 136               2515   
2             198              3263                 151               3414   
3              88                 0                  81                 81   
4              17                 0                   0                  0   

   Male_Detenues  Female_Detenues  Total_Detenues  Male_Others  Female_Others  \
0              4                0               4           22              0   
1              0                0               0            0              0   
2              0                0               0            0              0   
3              0                1               1            0              0   
4              0                0               0            0              0   

   Total_Others  Total_Male  Total_Female  Grand_Total  
0            22        6116           117         6233  
1             0        2673           137         2810  
2             0        3451           161         3612  
3             0           0           170          170  
4             0          17             0           17

Podemos decir que es una base de datos que nos muestra por medio del año en que realizó un censo, el tipo de carcel y el Estado donde se encuentra ubicado, la información de los prisioneros de la India, tanto de hombres como mujeres.
Registra si se encuentran detenidos, están bajo fianza o bajo juicio, también nos muestra el total de ocupantes.

### Ejercicio 3

#### Queremos tener de manera más puntual los nombres de las columnas que se tuvieron en cuenta

In [10]:
data.columns

Index(['State_UT_name', 'Year', 'Jail_Type', 'Male Convicts',
       'Female_Convicts', 'Total_Convicts', 'Male_Under_trial',
       'Female_Under_trial', 'Total_Under_trial', 'Male_Detenues',
       'Female_Detenues', 'Total_Detenues', 'Male_Others', 'Female_Others',
       'Total_Others', 'Total_Male', 'Total_Female', 'Grand_Total'],
      dtype='object')

In [9]:
data.shape[1]

18

Ahora si conocemos todos los elementos que se tuvieron en cuenta para la realización de este censo. En total fueron 18 columnas

### Ejercicio 4

#### Queremos saber durante cuántos años se realizaron los censos y cuántos en total se realizaron 


In [4]:
data.shape[0]

4023

In [5]:
data["Year"].nunique()

13

En la India durante 13 años, se realizaron 4023 censos para obtener información sobre los prisioneros.

### Ejercicio 5
#### Queremos analizar qué datos se obtuvieron en el primer censo que se realizó

In [6]:
data.head(1)

State_UT_name  Year     Jail_Type  Male Convicts  Female_Convicts  \
0  Andhra Pradesh  2001  Central Jail           3372                3   

   Total_Convicts  Male_Under_trial  Female_Under_trial  Total_Under_trial  \
0            3375              2718                 114               2832   

   Male_Detenues  Female_Detenues  Total_Detenues  Male_Others  Female_Others  \
0              4                0               4           22              0   

   Total_Others  Total_Male  Total_Female  Grand_Total  
0            22        6116           117         6233

Se puede decir que el primer censo se realizó en el año 2001, en el Estado de Andhra Pradesh en un tipo de carcel Central Jail.                   En el cual se reportó una diferencia extrema entre los numeros de convictos hombres con los convictos mujeres, siendo esta aproximadamente el 0.01% del numero de hombres.
También podriamos decir que en general los datos obtenidos en esta carcel, demuestran que hay una diferencia muy marcada entre el registro de hombres y mujeres, esto de podría deber a que en ese Estado o esa zona, no hay muchas habitantes mujeres o las mujeres no cometen tantos delitos. 
Al momento de realizar este censo, se encontraban 6233 ocupantes.

### Ejercicio 6

#### Queremos saber los tipos de las carceles donde se llevaron a cabo los censos

In [7]:
data["Jail_Type"].unique()

array(['Central Jail', 'District Jail', 'Sub_Jail', 'Women Jail',
       'Borstal School', 'Open Jail', 'Special Jail', 'Others', 'Total'],
      dtype=object)

Se tuvieron en cuenta para realizar esta recolección de datos, los tipos de carceles Centrak Jail, Dristrict Jail, Sub Jail, Women Jail, Borstal School, Open Jail, Special Jail y Others, siendo este último carceles que no se clasifican entre las anteriores.

### Ejercicio 7

#### Queremos saber el número total de hombres convictos por año en toda la India 

In [8]:
data[["Year","Male Convicts"]].groupby(["Year"]).sum() 

Male Convicts
Year               
2001         146948
2002         158918
2003         176792
2004         190338
2005         210058
2006         225542
2007         232114
2008         237828
2009         238706
2010         242314
2011         247266
2012         245552
2013         248526

Podemos observar que en la India, al 2001 hubo 146948 hombres convictos y al 2013 hubo 248526, es decir que podemos notar que el año y el número de prisioneros, año tras año aumenta de manera proporcional, sin embargo no significa que sera algo que siempre deba ser así porque muchos factores influyen en esta, pero en este caso podemos decir que se nota una proporción. 

### Ejercicio 8

#### Queremos saber el número total de mujeres convictas por año en toda la India 

In [9]:
data[["Year","Female_Convicts"]].groupby(["Year"]).sum() 

Female_Convicts
Year                 
2001             4378
2002             5324
2003             6740
2004             6716
2005             7086
2006             7808
2007             8116
2008             8786
2009             9176
2010             9264
2011             9918
2012            10026
2013            10690

Como podemos observar es un promedio bajo de mujeres convictas durante cada año, ya que teniendo en cuenta que en el año 2001, 4378 mujeres lo estuvieron y que el año que mayor número tuvo fue en el 2013 siendo estas 10690. Aproximadamente solo el 2.8% de la cantidad de convictos hombres.
La tabla también refleja resultados ascendentes, esto significa que en promedio cada año incrementa el número de convictas en las carceles, sin embargo es un incremento mínimo, con un rango aproximado de 946 en cada año.

### Ejercicio 9

#### Queremos saber a lo largo de los 13 años, cuantos convictos incluyendo hombres y mujeres hubo en la India

In [10]:
data["Female_Convicts"].sum() 
data["Male Convicts"].sum()

2800902

In [11]:
data["Female_Convicts"].sum()


104028

Teniendo en cuenta los dos registros, se dice que durante los 13 años 2.800.902 personas pasaron como convictos incluyendo hombres y mujeres, siendo 104.028 de estas, mujeres. Representando solo el 3,71% de los convictos durante todo el censo.

### Ejercicio 10

#### Queremos saber el promedio de personas que a lo largo de los 13 años han estado bajo juicio en la India


In [12]:
data["Total_Under_trial"].sum().mean()

6268960.0

Con esta información podemos determinar también un promedio por año, lo cual significaría un promedio de 522.413 personas que anual estuvieron bajo juicio en la India 

### Ejercicio 11

#### Queremos saber que ocupantes en total durante los 13 años que se realizó el censo, hubo en cada tipo de carcel en el estado de A & N Islands y sacar alguna conclusión

In [13]:
data[["State_UT_name","Grand_Total", "Jail_Type"]].groupby(["State_UT_name","Jail_Type"]).sum().head(9)

Grand_Total
State_UT_name Jail_Type                  
A & N Islands Borstal School            0
              Central Jail              0
              District Jail          6229
              Open Jail                 0
              Others                   30
              Special Jail            346
              Sub_Jail                140
              Total                  6745
              Women Jail                0

Con esta información podemos concluir que en el Estado de A & N Islands, no se encuentran tipos de carcel Borstal School, Central Jail, Open Jail, Women Jail, ya que durante 13 años no se registró ningun ocupante en esas categorías.


### Ejercicio 12

#### Queremos saber cuantos presos quedaron registrados en el año 2001 en el Estado de Andhra Pradesh sin tener el cuenta el tipo de carcel

In [14]:
data[(data["Year"] == 2001) & (data ["State_UT_name"] == "Andhra Pradesh")].shape [0]

9

Se puede decir que el 2001 fue un año donde el Estado de Andhra Pradesh no se presentaron muchos arrestos y podríamos decir que la población tiene buen comportamiento, o que es un Estado pequeño y por eso no hay tantos habiantes.

### Ejercicio 13

#### Cuál fue la número máximo de mujeres detenidas por otra razón que recibió la carcel de tipo para mujeres en el año 2002  


In [15]:
data[["Female_Others","Jail_Type","Year"]].groupby(["Year", "Jail_Type"]).max().head(18)

Female_Others
Year Jail_Type                    
2001 Borstal School              0
     Central Jail               30
     District Jail               4
     Open Jail                   0
     Others                      0
     Special Jail              143
     Sub_Jail                  832
     Total                    1441
     Women Jail                436
2002 Borstal School              0
     Central Jail               32
     District Jail               0
     Open Jail                   0
     Others                      0
     Special Jail              123
     Sub_Jail                  528
     Total                    1366
     Women Jail                683

En el 2002, 683 mujeres fueron detenidas por otras razones en carceles solo para mujeres, incrementando 247 registros desde el año 2001. Un incremento considerable porque siempre hemos visto que el número de mujeres es menor al de los hombres.

### Ejercicio 14

#### Queremos mirar en cuántos Estados se realizó el censo y para eso agruparemos las categorias según su frecuencia 

In [16]:
data["State_UT_name"].describe()

count          4023
unique           36
top       Karnataka
freq            117
Name: State_UT_name, dtype: object

Podemos observar que el censo se realizó en 36 Estados de la India, con un total de 4023 registros, por lo cual podríamos determinar un promedio de 111.75 registros por Estado, lo cual nos gustaría probar para saber un número más exacto por Estado.

### Ejercicio 15

#### Queremos confirmar la conclusión anterior y saber cuántos registros por Estado se realizaron

In [17]:
data["State_UT_name"].value_counts()

Karnataka            117
Uttaranchal          117
Madhya Pradesh       117
Meghalaya            117
Lakshadweep          117
Himachal Pradesh     117
Goa                  117
Nagaland             117
Kerala               117
Punjab               117
Uttar Pradesh        117
Daman & Diu          117
Orissa               117
Maharashtra          117
Tripura              117
West Bengal          117
D & N Haveli         117
A & N Islands        117
Manipur              117
Tamilnadu            117
Delhi                117
Jharkhand            117
Andhra Pradesh       117
Rajasthan            117
Bihar                117
Sikkim               117
Haryana              117
Mizoram              117
Gujarat              117
Chandigarh           117
Assam                117
Jammu & Kashmir      117
Pondicherry          117
Chhattisgarh         108
Arunachal Pradesh     45
Chhatisgarh            9
Name: State_UT_name, dtype: int64

In [18]:
data["State_UT_name"].value_counts().unique()

array([117, 108,  45,   9])

117, 108, 45 y 9 fueron las unicas variables de los numeros de censos que en cada Estado se realizaron. No quiere decir que solo se realizaron 117, 108, 45 y 9 censos, no. Sino que fueron los unicos valores que el archivo reconoce como asignados a los Estados.
Y teniendo en cuenta la tabla anterior, podemos ver que en el Estado de Chhatisgash solo se realizaron 9 pruebas, en Arunachal Pradesh 45 y en Chhattisash 108, por lo cual podemos determinar que en cada uno de los 33 Estados restantes se realizaron 117 censos.


### Ejercicio 16

#### Seguiremos comprobando la conclusión del ejercicio 13

In [19]:
data["State_UT_name"].value_counts().mean()

111.75

In [20]:
data["State_UT_name"].value_counts().std()

21.33859682091853

Confrimamos que el promedio de censos por Estado es de 111.75 con una desviación estandar de 21.34 lo cual nos permite afirmar que los datos se encuentran bien ya que el registro máximo por Estado fue de 117, estando dentro del rango de la desviación.

### Ejercicio 17

####  Queremos saber el número máximo registrado durante esos 13 años de hombres que estuvieron bajo juicio en la India

In [21]:
data["Male_Under_trial"].max()

57017

En la India entre el 2001 y el 2013, el máximo de hombres que estuvieron bajo juicio en un mismo año y en un mismo tipo de carcel fue de 57.017 

### Ejercicio 18

#### Queremos determinar el número de ocupantes que se presentaron durante cada año en las carceles de la Indía

In [26]:
data[["Year","Grand_Total"]].groupby(["Year"]).sum().head(13)

Grand_Total
Year             
2001       627270
2002       644714
2003       653038
2004       662782
2005       716736
2006       746542
2007       752792
2008       769506
2009       753938
2010       737996
2011       745852
2012       770270
2013       823984

Podemos observar que cada año si se presenta un incremento en el total de ocupantes, solo en el año 2009 hubo una disminución de 15.568 personas a comparación del año anterior.